Logic/ExtSign.hs

```diff
@@ -65,6 +66,17 @@ ext_signature_union l e1@(ExtSign s1 _) e2@(ExtSign s2 _) = do
     s <- signature_union l s1 s2
     return $ makeExtSign l s
 
+ext_signature_intersect :: Logic lid sublogics
+        basic_spec sentence symb_items symb_map_items
+        sign morphism symbol raw_symbol proof_tree
+        => lid -> ExtSign sign symbol -> ExtSign sign symbol
+               -> Result (ExtSign sign symbol)
+ext_signature_intersect l e1@(ExtSign s1 _) e2@(ExtSign s2 _) = do
+    checkExtSign l "u1" e1
+    checkExtSign l "u2" e2
```

Comment:
please use some more informative messages instead of "u1" and "u2". E.g. "left-hand side of intersection" and "right-hand side of intersection".